taskwiki/main.py

```diff
@@ -123,6 +132,34 @@ def info(self):
                 util.show_in_split(out, name='info', activate_cursorline=True)
             break  # Show only one task
 
+    @errors.pretty_exception_handler
+    def open(self):
+        compatible_annotation_found = False
+        for vimwikitask in self.tasks:
+            for annotation in vimwikitask.task["annotations"]:
+                annotation = annotation["description"]
+                proc = subprocess.Popen(["xdg-open", annotation])
```

Comment:
I don't think trying `xdg-open` for all annotations regardless of their content is wise.
Consider checking for file existence and matching a URI regexp. This will also make the wait unnecessary and will make it easier to do this robustly/portably, see also https://github.com/vimwiki/vimwiki/blob/619f04f89861c58e5a6415a4f83847752928252d/autoload/vimwiki/base.vim#L264.

Reply:
The new approach uses taskopen itself, so this will not be an issue.